ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -3911,20 +3962,27 @@ public boolean dropPartition(String dbName, String tableName, List<String> parti
    * @param tbl The table containing the partitions.
    * @param expr A serialized expression for partition predicates.
    * @param conf Hive config.
-   * @param result the resulting list of partitions
+   * @param partitions the resulting list of partitions
    * @return whether the resulting list contains partitions which may or may not match the expr
    */
   public boolean getPartitionsByExpr(Table tbl, ExprNodeGenericFuncDesc expr, HiveConf conf,
-      List<Partition> result) throws HiveException, TException {
-    assert result != null;
+      List<Partition> partitions) throws HiveException, TException {
+
+    assert partitions != null;
```

Comment:
May be use Preconditions.checkNotNull(partitions) here since assert is not guaranteed to be enabled always.

Reply:
Done